### ISSUE TITLE
Problem with bundler git gems

### ISSUE THREAD

**Issue by dallisonhcom** [jruby/warbler#165]
When I execute java -jar <JAR NAME> I get "LoadError: no such file to load" when I require any gems that are bundled from git.  

When I run the same ruby code via bundle exec it works just fine.  

Is this a bug or something Im doing wrong?

I'm using warbler (1.4.0.dev), bundler (1.3.5) and jruby-1.7.0
##

**Comment by blueboxjesse** [comment]
We ran into this today and documented much of it here (https://github.com/jruby/warbler/issues/235).  The bundle package --all fix the issue for us.

**Comment by aplumb-fiksu** [comment]
For me, camertron's solution of adding bundler/gems/** lib paths to the load path worked.

**Comment by the-michael-toy** [comment]
Trying to update our app to Jruby9000 and warbler, we also ran across this. There is a deeper look at why things fail including a small test at https://github.com/the-michael-toy/app_that_cant -- I took a lot of time to write this up, I hope this is helpful to someone,

TL;DR -- Our workaround is a little more focused, we look through the git-sources-gems in the jar and manually add their require_paths to the load path.

TL;DR#2 -- I think this is a bug in bundler or in the JRuby runtime that bundler somehow tripped on, but I couldn't get an traction on finding out more.

Here's the workaround, which was inspired by @camertron above, but I believe this is more correct, but I didn't test it with gems with require_paths to be sure ...

```Ruby
Dir.glob(File.join(ENV['GEM_HOME'], "bundler/gems/*/*.gemspec")).each do |spec_file|
  if (spec = Gem::Specification.load(spec_file))
    (spec.require_paths || ["lib"]).each { |gem_lib| $LOAD_PATH << File.join(File.dirname(gem_dir), gem_lib) }
  end
end
```

**Comment by kares** [comment]
would be great if someone came up with a PR that fixes this (keeping tests :green_heart:) instead of patching.
esp. if you're looking deep down the rabbit hole already ... there's no dedicated support on the project :open_mouth:

**Comment by the-michael-toy** [comment]
FWIW, I don't think this is a warbler issue, I think bundler is failing to do it's work from inside of a jar, but the gems are all laid out in the jar correctly.

**Comment by the-michael-toy** [comment]
I have re-run my test-app with warbler 2.0.4 and bundler 1.13.6 and `require 'bundler/setup'` is now correctly setting the load path for by git sourced jars. I took the workaround out of my app.
